Exercise/compute_lowest_terms.py

```diff
@@ -1,40 +1,73 @@
 # TODO: complete the function "lowest_terms" below
 
 def lowest_terms(x):
+	"""
+	Reducing fractions basically requires finding factors and dividing by them.
+	Containers (lists) are created e.g cont, factors and results to hold these
+	generated values.
+	"""
 	x = x.split('/')
-	l = []
-	fac = []
-	res = []
+	cont = []
+	factors = []
+	result = []
+
+	# count identifies the number of negatives in the fraction to determine its sign
 	count = 0
+
+	# Separating negatives from values
 	for i in x:
 		if '-' in i:
 			x[x.index(i)] = i[1:]
 			count += 1
+
+	# Converting strings to integers
 	for j in x:
-		l.append(int(j))
-	copy = l[:]
+		cont.append(int(j))
+
+	# copy is a copy of the prepared input that wouldn't be altered
+	copy = cont[:]
+
+	"""
+	Only the factors of the minimum value of either numerator or denominator are required
+	because any number greater than the minimum value wouldn't be a common factor for both
+	numerator and denominator and thus wouldn't be able to divide both. Thus we loop
+	through the minimum value 
+	"""
+	for i in range(min(cont)):
+		if min(cont) % (i + 1) == 0:
+			factors.append(i + 1)
+
+	# Reversing the list allows for dividing by larger common factors first before smaller ones
+	factors.reverse()
 
-	for i in range(min(l)):
-		if min(l) % (i + 1) == 0:
-			fac.append(i + 1)
-	fac.reverse()
+	"""
+	loop through the factors of the minimum values and reduce the fraction by common 
+	factors. This is done in-place so that the numerator and denominators are divided
+	till it is no longer possible
+	"""
+	for i in factors:
+		if min(cont) % i == 0 and max(cont) % i == 0:
```

Comment:
if cont[0] % i == 0 and cont[1]% i == 0:

cont only has two elements at index 0 and 1; calling min and max is another function call that is unnecessary (try to make your code efficient)

Reply:
Thank you, I over analysed that